apps/www/_blog/2023-04-10-o11y-day.mdx

```diff
@@ -0,0 +1,201 @@
+
+```bash
+curl "https://logflare.app/endpoints/query/69425db0-1cfb-48b4-84c7-2a872b6f0a61" \
+ -H 'Content-Type: application/json; charset=utf-8' \
+ -G -d "day=30"
+```
+
+Which gives us the response with the top org wide contributors for the last 30 days!
+
+```json
+{
+  "result": [
+    {
+      "count": 23404,
+      "gh_user": "vercel[bot]"
+    },
+    {
+      "count": 10005,
+      "gh_user": "joshenlim"
+    },
+    {
+      "count": 7026,
+      "gh_user": "MildTomato"
+    },
+    {
+      "count": 6405,
+      "gh_user": "fsansalvadore"
+    },
+    {
+      "count": 5195,
+      "gh_user": "saltcod"
+    },
+    {
+      "count": 3454,
+      "gh_user": "alaister"
+    },
+    {
+      "count": 2691,
+      "gh_user": "kevcodez"
+    },
+    {
+      "count": 2117,
+      "gh_user": "gregnr"
+    },
+    {
+      "count": 1769,
+      "gh_user": "Ziinc"
+    },
+    {
+      "count": 1749,
+      "gh_user": "chasers"
+    },
+    {
+      "count": 1430,
+      "gh_user": "Isaiah-Hamilton"
+    },
+    ...
+  ]
+}
+```
+
+We can then configure this Endpoint to cache results for a set interval of 10 minutes after the first API request, and proactively update those cached results every 2 minutes. This would result in 5 separate queries spread out áo cross the 10 minute cache time. Even if we were to throw tens of thousands of API requests at the endpoint, we would only sustain the cost of 5 queries to update the cached data.
+
+The initial request is fast because Logflare also performs setup (such as partitioning) on our BigQuery tables appropriately. Subsequent requests made are extremely fast as they get cached in-memory.
+
+The best part is that all these knobs can be tweaked based on the use case. If we have a hard realtime requirement, we can completely disable caching, or drop the proactive caching to update on a per-second basis.
+
+## The Self-hosted Challenge
+
+In order to execute the license change, we had to get rid of all closed-source dependencies. Prior to our efforts, Logflare relied on the closed source [General SQL Parser](https://www.sqlparser.com/) under a business licenses. However, it would not be an acceptable compromise under the Apache License.
+
+This resulted in an effort to switch Logflare over to an open source alternative, and we settled on the rust-based [sqlparser-rs](https://github.com/sqlparser-rs/sqlparser-rs) library, and also ended up helping to contribute a [few fixes](https://github.com/sqlparser-rs/sqlparser-rs/pulls?q=is%3Apr+is%3Amerged+author%3AZiinc) upstream for the BigQuery dialect.
+
+Besides the parser, a big part of getting Logflare working with Supabase locally was transforming the multi-tenant architecture into something that was self-hosting friendly and could be easily configurable. We also moved towards environment variable based configuration instead of compile-time configurations, and also exposed the Endpoints configurations necessary for Supabase Logs.
```

Comment:
```suggestion
Along with the parser, we invested a lot of effort into transforming the multi-tenant architecture into something that was self-hosting friendly and easily configurable. We moved towards environment variable based configuration instead of compile-time configurations, exposing the Endpoints configurations necessary for Supabase Logs.
```